signer/src/transaction_coordinator.rs

```diff
@@ -2822,24 +2799,18 @@ mod tests {
 
     /// Check that we skip processing bitcoin blocks if the chain tip in
     /// the state doesn't match the block hash passed in.
-    ///
-    /// Note: this test is a little sensitive to the current logic that
-    /// checks for smart contract deployment after checking whether the
-    /// chain tip is up to date.
     #[tokio::test]
     async fn should_skip_processing_bitcoin_blocks_if_not_coordinator() {
         let mut rng = testing::get_rng();
         let ctx = TestContext::builder()
             .with_in_memory_storage()
             .with_mocked_clients()
+            .modify_settings(|settings| {
+                // If we are the coordinator then we will wait for 10 seconds.
+                settings.signer.bitcoin_processing_delay = Duration::from_secs(10);
```

Comment:
Maybe add a little more context to the comment here from the comment below (i.e. explaining that this is to ensure that the short timeout further down actually times out). Reading top-down I started this comment with "do we really need 10s?!" 😅